staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1144,6 +1144,7 @@ message ContainerStatus {
     // image ID
     string image_ref = 9;
     // Brief CamelCase string explaining why container is in its current state.
+    // Must be set to OOMKilled for containers terminated by the Out-of-Memory killer.
```

Comment:
There are two sort of different OOM killers in the kernel; there is the global OOM killer and then there is cgroup-based OOM through the cgroup memory controller. Containerd uses the memory cgroup's oom event fd, would only know cgroup-based OOM. I believe CRI-O does the same. cc/ @mrunalp to confirm.

In this case, can we explicitly state here: "terminated by cgroup-based Out-of-Memory kiler" to avoid the confusion?

Reply:
> I believe CRI-O does the same.

correct!

Reply:
Makes sense, updated the comment accordingly. PTAL.